### ISSUE TITLE
stripzip should also strip access time, uid and gid

### ISSUE THREAD

**Issue by dimaqq** [Code0x58/python-stripzip#1]
Thanks for writing this tool!

For me, stripzip (as installed from pypi) resets file modification time (yay!) but leaves access time, uid and gid attributes.

Here's an example from MacOS, diff between 2 runs of (zip + stripzip):

```
--- details.1s.txt	2018-11-14 17:29:46.000000000 +0900
+++ details.2s.txt	2018-11-14 17:30:32.000000000 +0900
@@ -15,7 +15,7 @@
 0028   Length              0009
 002A   Flags               '03 mod access'
 002B   Mod Time            5BE94512 'Mon Nov 12 18:17:06 2018'
-002F   Access Time         5BEBDCB6 'Wed Nov 14 17:28:38 2018'
+002F   Access Time         5BEBDD12 'Wed Nov 14 17:30:10 2018'
 0033 Extra ID #0002        7875 'ux: Unix Extra Type 3'
 0035   Length              000B
 0037   Version             01
@@ -41,7 +41,7 @@
 006F   Length              0009
 0071   Flags               '03 mod access'
 0072   Mod Time            5BE3EA30 'Thu Nov  8 16:48:00 2018'
-0076   Access Time         5BEBDCB6 'Wed Nov 14 17:28:38 2018'
+0076   Access Time         5BEBDD12 'Wed Nov 14 17:30:10 2018'
 007A Extra ID #0002        7875 'ux: Unix Extra Type 3'
 007C   Length              000B
 007E   Version             01
```

Access times change because zip read the file, and uid/gid would depend on who ran zip.

**Comment by dimaqq** [comment]
Example dump with ID's:

```
03CB LOCAL HEADER #3       04034B50
03CF Extract Zip Spec      14 '2.0'
03D0 Extract OS            00 'MS-DOS'
03D1 General Purpose Flag  0000
     [Bits 1-2]            0 'Normal Compression'
03D3 Compression Method    0008 'Deflated'
03D5 Last Mod Time         00210000 'Tue Jan  1 00:00:00 1980'
03D9 CRC                   57570410
03DD Compressed Length     000006C1
03E1 Uncompressed Length   00001BC8
03E5 Filename Length       000F
03E7 Extra Length          001C
03E9 Filename              'somemodule/somefile.py'
03F8 Extra ID #0001        5455 'UT: Extended Timestamp'
03FA   Length              0009
03FC   Flags               '03 mod access'
03FD   Mod Time            5BE27B08 'Wed Nov  7 14:41:28 2018'
0401   Access Time         5BEBDCB6 'Wed Nov 14 17:28:38 2018'
0405 Extra ID #0002        7875 'ux: Unix Extra Type 3'
0407   Length              000B
0409   Version             01
040A   UID Size            04
040B   UID                 000001F5
040F   GID Size            04
0410   GID                 00000014
0414 PAYLOAD
```

I guess the entire "extended timestamp" should be nuked.

I can do `zip -X` for new archives, but I think stripzip would still be great for existing archives.

**Comment by Code0x58** [comment]
Thanks @dimaqq for the heads up and dumps, I missed that as the files I was was working with with were made by [zipfile](https://docs.python.org/3/library/zipfile.html) in setuptools.  

The core ZIP format doesn't include things like access times, ownership, or permissions. Information like that comes from additional data put into the _extra field_ on records, where it looks like there are at least [51 types of extension](http://fileformats.archiveteam.org/wiki/ZIP#Extensible_data_fields). In this case, a couple of extra fields types are involved:
 * _**0x5455**_ - extended timestamp information, fields of this type can be flat out dropped
 * _0x5855/0x7855/**0x7875**_ - additional Unix information that I haven't yet found the format of, but suspect it can't be dropped (without breaking symlinks), but it could probably be zeroed

Plan of action:
 * [ ] make issues on GitHub for known issues and work below
  * version/system information not zeroed
 * [ ] look into the the additional Unix information
 * [ ] make a function which can wipe out targeted extended fields (i.e. 0x5455)
  * always remove 0x5455 extra fields by default
 * [ ] document things like trying to get the zip software to avoid as much cleaning work as possible:
   - use constant compression ratio
   - avoid extension fields where possible, e.g. `zip --no-extra …`

In the mean time, it seems like rezipping with the `--no-extra` flag is as good as it gets without using an alternative tool. I'll try to get some time to work on these issues over the next week.

**Comment by Code0x58** [comment]
Oops, I can't operate a mouse

**Comment by Code0x58** [comment]
A slight delay... but it sounds like this is fixed in 1.0.0 @dimaqq
